module/applications/compendium-browser.mjs

```diff
@@ -181,12 +228,15 @@ export default class CompendiumBrowser extends HandlebarsApplicationMixin(Applic
    */
   async _prepareResultsContext(context, options) {
     const currentFilters = this.currentFilters;
+    const filterDefinitions = CONFIG[currentFilters.documentClass].dataModels[currentFilters.types?.first()]
```

Comment:
Was intended to be a shortcut to avoid re-calculating the filter intersections because if a filter is in the first type it is guaranteed to be in all types, but I realized that causes a problem if you set a filter and then change types. I've made a change to calculate the filter intersection once in `_prepareContext` instead.